src/twisted/web/http_headers.py

```diff
@@ -152,7 +151,21 @@ def removeHeader(self, name: AnyStr) -> None:
         """
         self._rawHeaders.pop(self._encodeName(name), None)
 
-    def setRawHeaders(self, name: AnyStr, values: Sequence[AnyStr]) -> None:
+    @overload
+    def setRawHeaders(self, name: Union[str, bytes], values: Sequence[bytes]) -> None:
```

Comment:
I don't understand this... but I don't know much about mypy and typing.

Maybe add a comment to explain why this extra typing definition is required.

I guess that the reason is something like this.

You can set a sequence of either str or bytes, but you can't set a sequence of mixed str and bytes.

Reply:
I'm also not sure I follow.  I don't think @adiroiban 's guess is correct because there's also a version for `Sequence[Union[str, bytes]]` below...

Reply:
Yes, the runtime behavior accepts any mix of str and bytes. The overloads were the only way I found to get it to not overconstrain `values`. The issue is that (IIRC, it's been a few weeks) MyPy doesn't accept e.g. `list[str]` as `Sequence[str | bytes]` and you can't express it with variance + a type variable because there is no base type other than object. So I wrote out all the valid cases in the tests and added overloads until MyPy accepted them.